src/differentials/abstract_zero.jl

```diff
@@ -43,21 +43,21 @@ Base.zero(::Type{<:AbstractDifferential}) = Zero()
     DoesNotExist() <: AbstractZero
 
 This differential indicates that the derivative does not exist.
-It is the differential for a Primal type that is not differentiable.
-Such an Integer, or Boolean (when not being used as a represention of a value that normally
-would be a floating point.)
-The only valid way to pertube such a values is to not change it at all.
-As such, `DoesNotExist` is functionally identical to `Zero()`,
-but provides additional semantic information.
+It is the differential for primal types that are not differentiable,
+such as integers or booleans (when they are not being used to represent
+floating-point values).
+The only valid way to perturb such values is to not change them at all.
+As a consequence, `DoesNotExist` is functionally identical to `Zero()`,
+but it provides additional semantic information.
 
-If you are adding this differential to a primal then something is wrong.
-A optimization package making use of this might like to check for such a case.
+If you are adding this differential to a primal, then something is wrong.
+An optimization package making use of this might want to check for such a case.
```

Comment:
I'm still not sure what this means. Is this something specific to `DoesNotExist()`, which does not apply to `Zero()`?

Reply:
Adding a `Zero()` to a primal during optimization is fine; it won't change anything but it is fine.
It means that for the function that was called that primal is never used.

Adding a `DoesNotExist()` to a primal during optimization means you are optimizing a variable that isn't even differentiable in any sense. Like a `String` or an index.

Reply:
Ah, now I get it. I've added a small sentence below to make this a bit more explicit.